src/Sparrow/ObjectPool.cs

```diff
@@ -274,4 +275,280 @@ private void Validate(object obj)
             }
         }
     }
+
+
+
+    /// <summary>
+    /// Generic implementation of object pooling pattern with predefined pool size limit. The main
+    /// purpose is that limited number of frequently used objects can be kept in the pool for
+    /// further recycling.
+    /// 
+    /// Notes: 
+    /// 1) it is not the goal to keep all returned objects. Pool is not meant for storage. If there
+    ///    is no space in the pool, extra returned objects will be dropped.
+    /// 
+    /// 2) it is implied that if object was obtained from a pool, the caller will return it back in
+    ///    a relatively short time. Keeping checked out objects for long durations is ok, but 
+    ///    reduces usefulness of pooling. Just new up your own.
+    /// 
+    /// Not returning objects to the pool in not detrimental to the pool's work, but is a bad practice. 
+    /// Rationale: 
+    ///    If there is no intent for reusing the object, do not use pool - just use "new". 
+    /// </summary>
+    public class ObjectPool<T, TFactoryParam> where T : class
```

Comment:
Do we really need to duplicate all this code? Can we have a single `ObjectPool<T, TFactoryParam>` and then a inherited class `ObjectPool<T>` that just specialize it?